src/main/java/seedu/address/logic/parser/UnmarkCommandParser.java

```diff
@@ -0,0 +1,56 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ID;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_INDEX;
+
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.UnmarkCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.StudentId;
+
+/**
+ * Parses input arguments and creates a new UnmarkCommand object
+ */
+public class UnmarkCommandParser implements Parser<UnmarkCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the UnmarkCommand and returns a UnmarkCommand
+     * object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public UnmarkCommand parse(String args) throws ParseException {
```

Comment:
Same as my earlier comment 🙂
```suggestion
    @SuppressWarnings("OptionalGetWithoutIsPresent") // This is checked using our custom function (arePrefixesPresent).
    public UnmarkCommand parse(String args) throws ParseException {
```